### PR TITLE
Adjust navbar

### PR DESCRIPTION
**Pull request by reecebisel**

### CODE REVIEW

**Comment by reecebisel** [comment]
Hey, this pull request is using my last branch instead of my master branch. They are at the same codewise I just submitted the wrong branch name.

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+class ArticlesController < ApplicationController
+  before_action :authenticate_user!, except: [:index]
```
Comment (elorest) [inline app/controllers/articles_controller.rb]: This line is unnecessary as petergate will handle all of that.

File: app/models/article.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Article < ApplicationRecord
+  belongs_to :user
+
+  validates :title, presence: true
+  validates :content, presence: true
+  validates :category, presence: true
+
+  scope :first_three_from_each_category, -> do
```
Comment (elorest) [inline app/models/article.rb:8]: Love that you wrote a scope. This is really inefficient though as it loads everything into memory and then groups it.

**Review by elorest — COMMENTED** [review]
Looks good. Checkout a couple of my comments.

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (reecebisel) [reply app/controllers/articles_controller.rb]: sweet. i'll remove that

File: app/models/article.rb (hunk 2 above)
Comment (reecebisel) [reply app/models/article.rb:8]: my refactor doesn't load everything into scope, but it does hit the DB four times to load things correctly. I'm not in love with this solution, but it seems slightly better than loading everything into memory

**Pull request closed without merging**